demo/tests/test_forms.py

```diff
@@ -776,3 +777,55 @@ def test_from_json_raised_error(self):
         self.assertEqual(len(context.exception.messages), 3)
         for message in context.exception.messages:
             self.assertEqual(message, 'This field is required.')
+
+
+class TestInnerPresets(TestCase):
+
+    def test_confirmation_not_required_field(self):
+        class TestPresets(FormidableForm):
+            number = fields.IntegerField(accesses={
+                'jedi': EDITABLE,
+            })
+            text = fields.CharField(accesses={
+                'jedi': EDITABLE,
+            })
+
+            class Meta:
+                presets = [
+                    ConfirmationPresets(
+                        [PresetArg(slug='left', field_id='number'),
+                         PresetArg(slug='right', value='42')],
+                    ),
+                    ConfirmationPresets(
+                        [PresetArg(slug='left', field_id='text'),
+                         PresetArg(slug='right', value='toto')],
+                    ),
+                ]
+
+        formidable = TestPresets.to_formidable(label='presets')
+        form_class = formidable.get_django_form_class(role='jedi')
+        form = form_class(data={'number': '', 'text': ''})
+        self.assertEqual(len(form.rules), 2)
+        self.assertTrue(form.is_valid(), form.errors)
+
+    def test_confirmation_required_fields(self):
+        class TestPresets(FormidableForm):
+            left = fields.IntegerField(accesses={
+                'jedi': REQUIRED,
+            })
+            right = fields.IntegerField(accesses={
+                'jedi': REQUIRED,
+            })
+
+            class Meta:
+                presets = [
+                    ConfirmationPresets(
+                        [PresetArg(slug='left', field_id='left'),
+                         PresetArg(slug='right', field_id='right')],
+                    ),
+                ]
+
+        formidable = TestPresets.to_formidable(label='presets')
+        form_class = formidable.get_django_form_class(role='jedi')
+        form = form_class(data={'left': '', 'right': ''})
+        self.assertFalse(form.is_valid(), form.errors)
```

Comment:
I don't understand the test ... The field are required so the form validation will be false.

I would like to see a test like that

=> Define a Validation Between two fields required for the Jedi
=> Get the dynamic_form_class for a padawan
=> Fill the form with empty value
=> validate the form (it has to be okay).